app/model/repository/BaseRepository.php

```diff
@@ -4,11 +4,16 @@
 
 use App\Exceptions\NotFoundException;
 use DateTime;
+use Doctrine\Common\Collections\Collection;
 use Doctrine\Common\Collections\Criteria;
 use Doctrine\ORM\EntityManagerInterface;
 use Doctrine\ORM\EntityRepository;
+use Doctrine\ORM\QueryBuilder;
 use Nette;
 
+/**
+ * @template T
```

Comment:
Smells like C++, yet we are in PHP.

Reply:
https://phpstan.org/blog/generics-in-php-using-phpdocs